ipatests/test_integration/test_sssd.py

```diff
@@ -148,3 +150,48 @@ def test_is_user_filtered(self, user):
             dp_req = ("Looking up [{0}] in data provider".format(
                 self.users[user]['name']))
             assert not dp_req.encode() in sssd_log
+
+    def test_external_group_paging(self):
+        """SSSD should fetch external groups without any limit.
+
+        This is regression test for https://pagure.io/SSSD/sssd/issue/4058
+        """
+        dn = DN(('cn', 'config'))
+        conn = self.master.ldap_connect()
+        entry = conn.get_entry(dn)  # pylint: disable=no-member
+        orig_limit = entry.single_value.get('nsslapd-sizelimit')
+        new_limit = int(orig_limit) + 2
+        tasks.kinit_admin(self.master)
+        try:
+            log_file = '{0}/sssd_{1}.log'.format(paths.VAR_LOG_SSSD_DIR,
+                                                 self.master.domain.name)
+            group_entry = '[{0}] external groups found'.format(new_limit)
+            logsize = tasks.get_logsize(self.master, log_file)
+            for i in range(new_limit):
+                self.master.run_command(['ipa', 'group-add', '--external',
+                                         'ext-ipatest{0}'.format(i)])
+            self.master.run_command(['id', self.users['ad']['name']])
+            sssd_logs = self.master.get_file_contents(log_file)[logsize:]
+            assert group_entry.encode() in sssd_logs
+            # set new-limit so that group-find will return actual count
```

Comment:
What is the point of the commands done after this line? The BZ is verified and the following lines only check that ipa group-find properly uses searchrecordslimit and nsslapd-sizelimit, but IMO it's not related to the BZ anymore.

Reply:
Yes this is just cross-check from ipa side the group count is correct or not.
If its not needed then I'll add only automation mentioned in https://github.com/freeipa/freeipa/pull/4077#discussion_r365776937